### PR TITLE
Tag Lambda functions with macro version

### PR DESCRIPTION
**Pull request by Czechh**
### What does this PR do?

With this change we tag the lambda functions built with the macro with the version used. This will allow us to get some usage tracking and help debug issues.

### Motivation

This change is part of the telemetry/usage OKR

### Testing Guidelines

- Unit tests
- Manual changes using SAM.

### Types of changes

- [ ] Bug fix
- [X] New feature
- [ ] Breaking change
- [] Misc (docs, refactoring, dependency upgrade, etc.)

### Check all that apply

- [X] This PR's description is comprehensive
- [ ] This PR contains breaking changes that are documented in the description
- [ ] This PR introduces new APIs or parameters that are documented and unlikely to change in the foreseeable future
- [ ] This PR impacts documentation, and it has been updated (or a ticket has been logged)
- [X] This PR's changes are covered by the automated tests
- [ ] This PR collects user input/sensitive content into Datadog

### CODE REVIEW

File: serverless/src/tags.ts [hunk 1]
```diff
@@ -31,3 +31,14 @@ export function addServiceAndEnvTags(lambdas: LambdaFunction[], service: string
     }
   });
 }
+
+export function addMacroTag(lambdas: LambdaFunction[], version: string | undefined) {
+  if (!version) return;
+
+  lambdas.forEach((lambda) => {
+    const tags = lambda.properties.Tags ?? [];
+    tags.push({ Value: `v${version}`, Key: "dd_sls_macro" });
```
Comment (nhinsch) [inline serverless/src/tags.ts:40]: Should `dd_sls_macro` be a constant?

File: serverless/test/index.spec.ts [hunk 2]
```diff
@@ -11,6 +11,7 @@ import {
 import { LogGroupDefinition } from "../src/forwarder";
 
 const LAMBDA_KEY = "HelloWorldFunction";
+const V_SEMVER_REGEX = /^v(0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*)(-(0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(\.(0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*)?(\+[0-9a-zA-Z-]+(\.[0-9a-zA-Z-]+)*)?$/;
```
Comment (nhinsch) [inline serverless/test/index.spec.ts:14]: What does `V_SEMVER` mean here? (I know semver refers to semantic version, but what's the "v" for?)

**Review by nhinsch — APPROVED** [review]
Lgtm, just a couple of minor nits

File: serverless/test/index.spec.ts (hunk 2 above)
Comment (Czechh) [reply serverless/test/index.spec.ts:14]: `v` is the prefix in the regex

File: serverless/src/tags.ts (hunk 1 above)
Comment (Czechh) [reply serverless/src/tags.ts:40]: Good call, was following the pattern for the other tags. Going to move those to constants maybe using an enum.

File: serverless/test/index.spec.ts (hunk 2 above)
Comment (nhinsch) [reply serverless/test/index.spec.ts:14]: Hmm, I would not have guessed that, I would just call this `VERSION_REGEX` or something, but up to you

**Pull request merged**